onnxruntime/core/framework/utils.cc

```diff
@@ -507,6 +507,48 @@ static common::Status CopyInputsAcrossDevices(const SessionState& session_state,
   return Status::OK();
 }
 
+#ifdef ORT_ENABLE_STREAM
+struct DeviceStreamCollectionHolder {
```

Comment:
>DeviceStreamCollectionHolder [](http://example.com/codeflow?start=7&length=28)

why we need a holder here?
the collection might get "recycled" accidentally?

Reply:
Lei just move this func to a new place. the original implementation is to use this holder as a guard to make sure if any error happened we return the device streams to session state correctly.